core/src/main/java/org/elasticsearch/plugins/Plugin.java

```diff
@@ -107,7 +112,9 @@
      */
     public Collection<Object> createComponents(Client client, ClusterService clusterService, ThreadPool threadPool,
                                                ResourceWatcherService resourceWatcherService, ScriptService scriptService,
-                                               NamedXContentRegistry xContentRegistry) {
+                                               NamedXContentRegistry xContentRegistry, Environment environment,
+                                               NodeEnvironment nodeEnvironment, NamedWriteableRegistry namedWriteableRegistry,
+                                               BiFunction<Settings, Path, Node> clientNodeBuilder) {
```

Comment:
I think we should not make this a first class citizen. Instead I suggest we remove the Node#newNode entirely and add a `MockTribePlugin` class that passes such a function that delegates to / creates a `MockNode`. Given that tribe goes away I think that is the right tradeoff. WDYT?